doc/contribute/index.rst

```diff
@@ -345,6 +345,43 @@ following exceptions:
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
 
+The C standard reserves all identifiers that begin with an underscore
+for implementation details.
+
+.. note::
+   According to the C Standard, 7.1.3 [ISO/IEC 9899:2011]
+
+   * All identifiers that begin with an underscore and either an
+     uppercase letter or another underscore are always reserved for any
+     use.
+   * All identifiers that begin with an underscore are always reserved
+     for use as identifiers with file scope in both the ordinary and tag
+     name spaces.
+   * Each macro name in any of the following subclauses
+     (including the
+     future library directions) is reserved for use as specified if any
+     of its associated headers is included,unless explicitly stated
+     otherwise.
+   * All identifiers with external linkage (including future library
+     directions) and errno are always reserved for use as identifiers
+     with external linkage.
+
+Zephyr contains the follow name conventions to
+address C standard violations:
+
+* Static functions should not contain underscore
```

Comment:
Actually, the functions exported to other parts of the Zephyr kernel are usually static ones. So the division into static and non-static should probably be 'exported' and 'internal' or similar wording? There was a good reasoning behind using static, was it to avoid symbol table issues?

Reply:
agree about changing to change to exported/internal. Answering about using static, there are several reasons:
* Declaring static the compiler can make optimizations making nline the functions.
* Locate the static functions close to the callers allowing a ‘short’ call rather than a ‘long’ call 
* Look at registers used by the local functions and thus only stack the required scratch registers rather than stacking all of the registers required by the compiler’s calling convention
* Limit the scope of the function